File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = [1]
    end_day = []

    for i in range(len(progresses)):
        left_work = 100 - progresses[i]
        finish_day = left_work // speeds[i]

        if left_work % speeds[i] > 0:
            finish_day += 1

        end_day.append(finish_day)

    day_cnt = end_day[0]
    day_idx = 0

    for i in range(1, len(end_day)):
        if end_day[i] <= day_cnt:
            answer[day_idx] += 1
        else:
            day_cnt = end_day[i]
            day_idx += 1
            answer.append(1)

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> progress_left_day;
    
    for(int i = 0; i<progresses.size(); i++){
        int left_day = (100 - progresses[i]) / speeds[i];
        if ((100 - progresses[i]) % speeds[i]){
            left_day++;
        }
        progress_left_day.push_back(left_day);
    }

    int stack_idx = -1;
    int progress_left = 0;
    
    for(int i = 0; i<progresses.size(); i++){
        if (progress_left < progress_left_day[i]){
            progress_left = progress_left_day[i];
            stack_idx++;
            answer.push_back(1);
        }
        else{
            answer[stack_idx]++;
        }
    }
    
    return answer;
}